kernelci/build.py

```diff
@@ -1211,6 +1233,46 @@ def install(self, verbose=False):
         return super().install(verbose)
 
 
+class FetchFirmware(Step):
+
+    @property
+    def name(self):
+        return 'firmware'
+
+    def run(self, jopt=None, verbose=False, opts=None):
+        """Fetch linux-firmware repository
+
+        If kernel have CONFIG_EXTRA_FIRMWARE enabled we need to fetch
+        fresh snapshot of linux-firmware git repository, otherwise
+        kernel will not be able to build
+        """
+        # CONFIG_EXTRA_FIRMWARE_DIR need absolute path
+        full_path = os.path.abspath(self._output_path)
+        fwdir = os.path.join(full_path, 'linux-firmware')
+        key = self._kernel_config_getkey('CONFIG_EXTRA_FIRMWARE')
+        if (key and len(key) < 3):
+            print("External firmware not required")
+            return False
+        if os.path.isdir(fwdir):
+            print("Firmware directory already present")
+        else:
+            print("Fetching firmware")
+            shell_cmd(f"""
+set -ex
+git clone --depth=1 \
+git://git.kernel.org/pub/scm/linux/kernel/git/firmware/linux-firmware.git \
+{fwdir}
+""")
```

Comment:
Can we use `kernelci.build.update_repo()` for this?

Reply:
As far as i can see it doesn't make shallow clone and looks a bit heavier. Probably it will slow down things significantly, but i will test and compare.

Reply:
Maybe add a `shallow=True` optional argument?

Reply:
Done